dds_web/api/files.py

```diff
@@ -276,18 +350,83 @@ def delete(self):
 
         check_eligibility_for_deletion(project.current_status, project.has_been_available)
 
-        with DBConnector(project=project) as dbconn:
-            not_removed_dict, not_exist_list, error = dbconn.delete_multiple(
-                files=flask.request.json
-            )
-
-            # S3 connection error
-            if not any([not_removed_dict, not_exist_list]) and error != "":
-                raise S3ConnectionError(str(error))
+        not_removed_dict, not_exist_list = self.delete_multiple(
+            project=project, files=flask.request.json
+        )
 
         # Return deleted and not deleted files
         return {"not_removed": not_removed_dict, "not_exists": not_exist_list}
 
+    def delete_multiple(self, project, files):
+        """Delete multiple files."""
+
+        not_removed_dict, not_exist_list = ({}, [])
+
+        with ApiS3Connector(project=project) as s3conn:
+
+            # Delete each file
+            for x in files:
+                # Delete from db
+                try:
+                    name_in_bucket = self.delete_one(project=project, filename=x)
```

Comment:
It should be `filename=files[x]` here as files is a dict and not a list.

Reply:
No `files` is a list. 
`flask.request.json` doesn't need to be a dict. In this case it's a list of files being passed in. 
```
flask.current_app.logger.debug(f"Files: {files}")
```
gives the following output
```
dds_backend   | [2022-02-13 18:34:14,535] files [DEBUG] Files: ['testfile.txt']
```

Reply:
👍